### PR TITLE
[action] Add github action to merge mssonicbld's PRs which can be merged

### PR DESCRIPTION
**Pull request by liushilongbuaa**
#### Why I did it
Scan mssonicbld's PR and merge those whose checks passed.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: .github/workflows/automerge_scan.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - name: Debug
+      env:
+        TOKEN: ${{ secrets.TOKEN }}
+      run: |
+        set -e
+
+        echo ${TOKEN} | gh auth login --with-token
+        gh pr list -R sonic-net/sonic-buildimage -A mssonicbld --json additions,assignees,author,baseRefName,body,changedFiles,closed,closedAt,comments,commits,createdAt,deletions,files,headRefName,headRepository,headRepositoryOwner,id,isCrossRepository,isDraft,labels,latestReviews,maintainerCanModify,mergeCommit,mergeStateStatus,mergeable,mergedAt,mergedBy,milestone,number,potentialMergeCommit,projectCards,reactionGroups,reviewDecision,reviewRequests,reviews,state,statusCheckRollup,title,updatedAt,url > prs.log
```
Comment (xumia) [inline .github/workflows/automerge_scan.yml:18]: The PR should be created by mssonicbld, right?

Comment (liushilongbuaa) [reply .github/workflows/automerge_scan.yml:18]: yes, -A == --author.

File: .github/workflows/automerge_scan.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            # EasyCLA finish flag:  state=SUCCESS
+            # Others finish flag:   status=COMPLETED
+            if [[ "$status" != "COMPLETED" ]] && [[ "$state" != "SUCCESS" ]];then
+              echo "$url Check didn't finished."
+              echo $check | jq
+              continue 2
+            fi
+            # EasyCLA success flag:  state=SUCCESS
+            # Others success flag:   conclusion in SUCCESS,NEUTRAL
+            if [[ "$state" != "SUCCESS" ]] && [[ "$conclusion" != "SUCCESS" ]] && [[ "$conclusion" != "NEUTRAL" ]];then
```
Comment (xumia) [inline .github/workflows/automerge_scan.yml:48]: Any of the condition failed, then continue?

File: .github/workflows/automerge_scan.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        set -e
+
+        count=$(cat prs.log | jq 'length')
+        for ((i=0;i<$count;i++))
+        do
+          url=$(cat prs.log | jq -r ".[$i].url")
+          echo PR: $(($i+1))/$count, URL: $url
+          [[ "$url" == "" ]] && continue
+          checks=$(cat prs.log | jq ".[$i].statusCheckRollup")
+          checks_count=$(echo $checks | jq 'length')
```
Comment (xumia) [inline .github/workflows/automerge_scan.yml:31]: The checks_count should be greater than 1, right?
Can we check the required checks?

File: .github/workflows/automerge_scan.yml (hunk 1 above)
Comment (xumia) [inline .github/workflows/automerge_scan.yml:18]: We should only take action for the PRs not merged in long time, for example, the PR createdAt >= <specific value>

File: .github/workflows/automerge_scan.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          echo Checks count: $checks_count
+          for ((j=0;j<$checks_count;j++))
+          do
+            check=$(echo $checks | jq ".[$j]")
+            status=$(echo $check | jq -r '.status')
+            state=$(echo $check | jq -r '.state')
+            conclusion=$(echo $check | jq -r '.conclusion')
+            # EasyCLA finish flag:  state=SUCCESS
+            # Others finish flag:   status=COMPLETED
+            if [[ "$status" != "COMPLETED" ]] && [[ "$state" != "SUCCESS" ]];then
```
Comment (xumia) [inline .github/workflows/automerge_scan.yml:41]: Any of the conditions does not satisfy, then continue?

Comment (xumia) [inline .github/workflows/automerge_scan.yml:41]:
```suggestion
            if [[ "$status" != "COMPLETED" ]] || [[ "$state" != "SUCCESS" ]];then
```

File: .github/workflows/automerge_scan.yml (hunk 2 above)
Comment (xumia) [inline .github/workflows/automerge_scan.yml:48]:
```suggestion
            if [[ "$state" != "SUCCESS" ]] || [[ "$conclusion" != "SUCCESS" ]] || [[ "$conclusion" != "NEUTRAL" ]];then
```

File: .github/workflows/automerge_scan.yml (hunk 3 above)
Comment (liushilongbuaa) [reply .github/workflows/automerge_scan.yml:33]: It is easier to check all checks instead of required checks.

**Review by xumia — APPROVED** [review]

**Comment by vivekrnv** [comment]
AutoMerge action is continuously failing in my private fork and causing noice. Any idea how to make these work. Eg: 
https://github.com/vivekrnv/sonic-buildimage/actions/runs/3680431978

**Comment by liushilongbuaa** [comment]
> AutoMerge action is continuously failing in my private fork and causing noice. Any idea how to make these work. Eg: https://github.com/vivekrnv/sonic-buildimage/actions/runs/3680431978

It seems that you need to disable it manually. https://github.com/orgs/community/discussions/26704

**Pull request merged**